x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java

```diff
@@ -493,6 +495,12 @@ public void resolveAsSeparateMappings(String indexWildcard, String javaRegex, bo
                 if (typeEntry.getKey().startsWith("_") && typeCap.getType().startsWith("_")) {
                     continue;
                 }
+                
+                // skip the "flattened" type of field's "_keyed" subfield
```

Comment:
I wonder if we cannot make this a bit generic by looking for field names that have leaves starting with `_` and if so, look at their type  - if its not recognized, mark the field as unsupported.